PencilCase/PencilCase/scripts/Pages/DataAnalyse.ts

```diff
@@ -1,58 +1,460 @@
 ﻿import { PageBase } from './PageBase';
 import { Navigator } from '../Navigator';
 import * as Consts from './Consts';
+import { OrderTypes } from '../Models/Order';
+import { OrderRepository } from '../Repositories/OrderRepository';
+import { ProductRepository } from '../Repositories/ProductRepository';
+
+declare var palette: any;
 
 export class DataAnalyse extends PageBase {
     private navigator: Navigator = Navigator.instance;
+    private isChartVisible: KnockoutObservable<boolean> = ko.observable(false);
+    private chartTimespanOption: ChartTimespanOptions;
+    private chartDataType: KnockoutObservable<string> = ko.observable("Quantity");
+    private quantitySaleTypeOptinos = [
+        { text: "Retail", value: OrderTypes.Retail },
+        { text: "Wholesale", value: OrderTypes.Wholesale },
+        { text: "Both", value: -1 }
+    ];
+    private selectedQuantitySaleType: KnockoutObservable<any> = ko.observable(OrderTypes.Retail);
+
+    private orderRepository = new OrderRepository();
+    private productRepository = new ProductRepository();
+
+    private chartComponent;
+    private chartPageTitle = ko.observable('');
+
+    private customStartDate: string = '';
+    private customEndDate: string = '';
+    private isDatePickersInitialized = false;
+
+    private isEmptyData: KnockoutObservable<boolean> = ko.observable(false);
 
     constructor() {
         super();
         this.title = ko.observable("Data Analyse");
         this.pageId = Consts.Pages.DataAnalyse.Id;
-        this.back = Navigator.instance.goHome;
+
+        this.selectedQuantitySaleType.subscribe((newValue: any) => {
+            this.refreshChart();
+        });
+
+        $('input[name="daterange"]').daterangepicker({
```

Comment:
This block of code need to removed, this is the code for testing, missed to remove.